api/restHandler/AppWorkflowRestHandler.go

```diff
@@ -219,10 +240,47 @@ func (impl AppWorkflowRestHandlerImpl) FindAppWorkflow(w http.ResponseWriter, r
 
 	workflows["appId"] = app.Id
 	workflows["appName"] = app.AppName
-	if len(workflowsList) > 0 {
-		workflows["workflows"] = workflowsList
+	if len(workflowsList) > 0 && app.AppType == helper.Job {
+		// RBAC
+		userEmailId, err := impl.userAuthService.GetEmailFromToken(token)
+		if err != nil {
+			impl.Logger.Errorw("error in getting user emailId from token", "err", err)
+			common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+			return
+		}
+
+		var workflowNames []string
+		var workflowIds []int
+		var updatedWorkflowList []appWorkflow.AppWorkflowDto
+		var rbacObjects []string
+		workNameObjectMap := make(map[string]appWorkflow.AppWorkflowDto)
+
+		for _, workflow := range workflowsList {
+			workflowNames = append(workflowNames, workflow.Name)
+			workflowIds = append(workflowIds, workflow.Id)
+		}
+		workflowIdToObjectMap := impl.enforcerUtil.GetAllWorkflowRBACObjectsByAppId(appId, workflowNames, workflowIds)
+		itr := 0
+		for _, val := range workflowIdToObjectMap {
+			rbacObjects = append(rbacObjects, val)
+			workNameObjectMap[val] = workflowsList[itr]
+			itr++
+		}
```

Comment:
Can be solved in v2 : this iteration is not needed, we can directly pass workflow objects to enforcer util method